src/classes/groundplaces.ts

```diff
@@ -0,0 +1,161 @@
+   * @returns {StopCluster}
+   */
+  // @ts-ignore
+  public createStopCluster(stopClusterInfos: StopCluster): StopCluster {}
+
+  /**
+   * @description Update the stopGroup with the new values given.
+   * @param {StopGroupGpuid} stopGroupGpuid - Ground place unique identifier of a StopGroup.
+   * @param {Properties} propertiesToUpdate - Properties that need to be update.
+   * @returns {StopGroup}
+   */
+  // @ts-ignore
+  public updateStopGroup(stopGroupGpuid: StopGroupGpuid, propertiesToUpdate: Properties): StopGroup {}
+
+  /**
+   * @description Update the stopCluster with the new values given.
+   * @param {StopClusterGpuid} stopClusterGpuid - Ground place unique identifier.
+   * @param {Properties} propertiesToUpdate - Properties that need to be update.
+   * @returns {StopCluster}
+   */
+  // @ts-ignore
+  public updateStopCluster(stopClusterGpuid: StopClusterGpuid, propertiesToUpdate: Properties): StopCluster {}
+
+  /**
+   * @description Remove a stopGroup from a stopCluster.
+   * @param {StopClusterGpuid} stopClusterGpuid - Ground place unique identifier of a StopCluster.
+   * @param {StopGroupGpuid} stopGroupGpuid - StopGroup Ground place unique identifier to remove.
+   * @returns {void}
+   */
+  public removeStopGroupFromStopCluster(stopClusterGpuid: StopClusterGpuid, stopGroupGpuid: StopGroupGpuid): void {}
+
+  /**
+   * @description Move a segmentProviderStop from a stopGroup to another stopGroup.
+   * @param {string} segmentProviderId - The identifier of the segmentProvider to move.
+   * @param {StopGroupGpuid} fromStopGroupGpuid - Ground place unique identifier of the old stopGroup.
+   * @param {StopGroupGpuid} intoStopGroupGpuid - Ground place unique identifier of the new stopGroup.
+   * @returns {void}
+   */
+  // WARNING: The segmentProviderStop cannot be without a parent.
+  public moveSegmentProviderStop(
+    segmentProviderId: string,
+    fromStopGroupGpuid: StopGroupGpuid,
+    intoStopGroupGpuid: StopGroupGpuid,
+  ): void {}
+
+  /**
+   * @description Move a stopGroup from a stopCluster to another stopCluster.
+   * @param {StopGroupGpuid} stopGroupToMoveGpuid - Ground place unique identifier of the stopGroup to move.
+   * @param {StopClusterGpuid} fromStopClusterGpuid - Ground place unique identifier of the old stopCluster.
+   * @param {StopClusterGpuid} intoStopClusterGpuid - Ground place unique identifier of the new stopCluster.
+   * @returns {StopCluster}
+   */
+  // @ts-ignore
+  public moveStopGroup(
+    stopGroupToMoveGpuid: StopGroupGpuid,
+    fromStopClusterGpuid: StopClusterGpuid,
+    intoStopClusterGpuid: StopClusterGpuid,
+  ): StopCluster {}
+
+  /**
+   * @description Delete place only if it's empty.
+   * @param {StopGroupGpuid|StopClusterGpuid} placeToRemoveGpuid - Ground place unique identifier of the place to remove.
+   * @returns {void}
+   */
+  public deletePlace(placeToRemoveGpuid: StopGroupGpuid | StopClusterGpuid): void {}
```

Comment:
maybe split it into 2 functions ? i think the logic will be different for stopGroups and stopCluster but i dont know.

Reply:
I'm ok with your suggestion, what do you think @Simobain ?

Reply:
Spec related https://www.notion.so/tictactrip/Package-de-manipulation-de-fichier-de-r-f-rence-edc6ef6e1956425fa22abf6b48408063#102fa628152248beacad30f6f8765d45

Reply:
Yes split it :)

Reply:
The inner logic of removing an empty place (meaning not related to anything is the same)

Reply:
However. It's better split them.

Reply:
![image](https://user-images.githubusercontent.com/4479510/98011687-7a8ba880-1df8-11eb-8382-5af5d0f72852.png)
I'll do a genric Method called deletePlace that takes a GPUID type in param. 
et two methods :  
* deleteStopGroup
* deleteStopCluster 
that uses the inner method.

Reply:
Done